Build fail when passing CONFIG_ string assignment from the command line

**Describe the bug**
Hey folks, looks like passing string configs from the command line broke with https://github.com/zephyrproject-rtos/zephyr/pull/85994

```console
$ west build -p -b nrf52dk/nrf52832 samples/hello_world/ -DCONFIG_BOOT_BANNER_STRING=\"broken\"
...
CMake Error at /home/fabiobaltieri/zephyrproject/zephyr/cmake/modules/yaml.cmake:552 (string):
  string sub-command JSON failed parsing json string: * Line 81, Column 134

    Missing ',' or '}' in object declaration

  .
```

added a print for `${json}` to figure what's going on and this is the problematic line:

```
    "command" : "/home/fabiobaltieri/.venv/bin/west build -p -b nrf52dk/nrf52832 samples/hello_world/ -DCONFIG_BOOT_BANNER_STRING=\\"broken\\"",
```

usual multiple escape issue it seems.

@tejlmand @pillo79 @nordicjm could any of you folks look into it?

Thanks!

**To Reproduce**
See above

**Expected behavior**
Build happily

**Impact**
It's pretty bad.

**Environment (please complete the following information):**
v4.1.0-rc2, Zephyr SDK, Linux, the usual

--- a/cmake/modules/extensions.cmake
+++ b/cmake/modules/extensions.cmake
@@ -3037,9 +3037,9 @@ endfunction()
 # This function extends the CMake string function by providing additional
 # manipulation arguments to CMake string.
 #
-# ESCAPE:   Ensure that any single '\' in the input string is escaped with the
-#           escape char '\'. For example the string 'foo\bar' will be escaped
-#           so that it becomes 'foo\\bar'.
+# ESCAPE:   Ensure that any single '\', except '\"', in the input string is
+#           escaped with the escape char '\'. For example the string 'foo\bar'
+#           will be escaped so that it becomes 'foo\\bar'.
 #           Backslashes which are already escaped will not be escaped further,
 #           for example 'foo\\bar' will not be modified.
 #           This is useful for handling of windows path separator in strings or
@@ -3084,7 +3084,7 @@ function(zephyr_string)
     # If a single '\' is discovered, such as 'foo\bar', then it must be escaped like: 'foo\\bar'
     # \\1 and \\2 are keeping the match patterns, the \\\\ --> \\ meaning an escaped '\',
     # which then becomes a single '\' in the final string.
-    string(REGEX REPLACE "([^\\][\\])([^\\])" "\\1\\\\\\2" work_string "${ZEPHYR_STRING_UNPARSED_ARGUMENTS}")
+    string(REGEX REPLACE "([^\\][\\])([^\\\"])" "\\1\\\\\\2" work_string "${ZEPHYR_STRING_UNPARSED_ARGUMENTS}")
   endif()
 
   set(${return_arg} ${work_string} PARENT_SCOPE)

